src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/DepedencyCollectorDiagnosticListener.cs

```diff
@@ -0,0 +1,103 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.DiagnosticListeners
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Globalization;
+    using System.Linq;
+    using System.Net.Http;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.Extensions.DiagnosticAdapter;
+
+    /// <summary>
+    /// <see cref="IApplicationInsightDiagnosticListener"/> implementation that listens for HTTP requests from this AspNetCore application.
+    /// </summary>
+    internal class DependencyCollectorDiagnosticListener : IApplicationInsightDiagnosticListener
+    {
+        private readonly TelemetryClient client;
+        private readonly ConcurrentDictionary<Guid, DependencyTelemetry> requestTelemetry = new ConcurrentDictionary<Guid, DependencyTelemetry>();
+
+        /// <inheritdoc/>
+        public string ListenerName { get; } = "HttpHandlerDiagnosticListener"; // This value comes from: https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/Common/src/System/Net/Http/HttpHandlerLoggingStrings.cs#L12
+
+        public DependencyCollectorDiagnosticListener(TelemetryClient client)
+        {
+            this.client = client;
+        }
+
+        /// <summary>
+        /// Diagnostic event handler method for 'System.Net.Http.Request' event.
+        /// </summary>
+        [DiagnosticName("System.Net.Http.Request")]
+        public void OnRequestSent(HttpRequestMessage request, Guid loggingRequestId, long timestamp)
+        {
+            string httpMethod = request.Method.Method;
+            Uri requestUri = request.RequestUri;
+            string resourceName = requestUri.AbsolutePath;
+            if (!string.IsNullOrEmpty(httpMethod))
+            {
+                resourceName = httpMethod + " " + resourceName;
+            }
+
+            DependencyTelemetry telemetry = new DependencyTelemetry();
+            telemetry.Start();
+            this.client.Initialize(telemetry);
```

Comment:
Consider using ```this.client.StartOperation<DependencyTelemetry>(resourceName)``` here. It does the ```telemetry.Start()``` call and ```this.client.Initialize```.
You get back an ```IDisposable``` operation holder which you should put in the dictionary (instead of ```DependencyTelemetry```) and ```Dispose``` in the ```OnResponseReceived``` handler.